### PR TITLE
remove Coordinator related code

### LINKED ISSUE #4: Remove milestone information from getNodeInfo
Remove the following fields:
* `latestMilestone`
* `latestMilestoneIndex`
* `latestSolidSubtangleMilestone`
* `latestSolidSubtangleMilestoneIndex`
* `milestoneStartIndex`
* `latestMilestoneIndex`
* `latestSolidSubtangleMilestone`
* `latestSolidSubtangleMilestoneIndex`
* `milestoneStartIndex`

### LINKED ISSUE #5: Change LedgerValidator to `return true;`
Remove all LedgerValidator logic

### LINKED ISSUE #7: Remove Snapshot class

### LINKED ISSUE #8: Remove Milestone class

### LINKED ISSUE #10: Bring GTTA back to life with naive `entryPoint` (start at genesis)

### PR DESCRIPTION
**Pull request by alon-e**
fixes: #7 
fixes: #8 
fixes: #10
fixes: #5
fixes: #4
removes all coo-related configuration vars

the code compiles and passed tests.

### CODE REVIEW

**Comment by alongalky** [comment]
Do you know why the Travis CI build fails?

File: src/main/java/com/iota/iri/LedgerValidator.java [hunk 1]
```diff
@@ -48,132 +46,8 @@ public LedgerValidator(Tangle tangle, MilestoneTracker milestoneTracker, Transac
...
-                    nonAnalyzedTransactions.offer(transactionViewModel2.getTrunkTransactionHash());
-                    nonAnalyzedTransactions.offer(transactionViewModel2.getBranchTransactionHash());
-                }
-            }
-        }
+        //TODO CLIRI: disabled diff
+        return new HashMap<>();
     }
 
     /**
```
Comment (alongalky) [inline src/main/java/com/iota/iri/LedgerValidator.java:14]: Delete the comment

File: src/main/java/com/iota/iri/LedgerValidator.java [hunk 2]
```diff
@@ -14,14 +14,12 @@
 
     private final Logger log = LoggerFactory.getLogger(LedgerValidator.class);
```
Comment (alongalky) [inline src/main/java/com/iota/iri/LedgerValidator.java]: All private fields throw "The value of the is not used" warnings. Recommend deleting all of them, we'll re-think this when we're finished with the z-net stage

File: src/main/java/com/iota/iri/LedgerValidator.java [hunk 3]
```diff
@@ -14,14 +14,12 @@
...
     private final Tangle tangle;
-    private final MilestoneTracker milestoneTracker;
     private final TransactionRequester transactionRequester;
     private final MessageQ messageQ;
     private volatile int numberOfConfirmedTransactions;
```
Comment (alongalky) [inline src/main/java/com/iota/iri/LedgerValidator.java]: This one especially should be removed

File: src/main/java/com/iota/iri/LedgerValidator.java [hunk 4]
```diff
@@ -14,14 +14,12 @@
...
 
-    public LedgerValidator(Tangle tangle, MilestoneTracker milestoneTracker, TransactionRequester transactionRequester, MessageQ messageQ) {
+    public LedgerValidator(Tangle tangle, TransactionRequester transactionRequester, MessageQ messageQ) {
         this.tangle = tangle;
-        this.milestoneTracker = milestoneTracker;
         this.transactionRequester = transactionRequester;
         this.messageQ = messageQ;
     }
```
Comment (alongalky) [inline src/main/java/com/iota/iri/LedgerValidator.java]: Update comment below: correct @params and remove milestone references

File: src/main/java/com/iota/iri/LedgerValidator.java [hunk 5]
```diff
@@ -14,14 +14,12 @@
```
Comment (alongalky) [inline src/main/java/com/iota/iri/LedgerValidator.java:10]: Line 3: `[Java] The import com.iota.iri.controllers is never used`

File: src/main/java/com/iota/iri/TransactionValidator.java [hunk 6]
```diff
@@ -133,12 +126,12 @@ public int getMinWeightMagnitude() {
...
-            return transactionViewModel.getTimestamp() < snapshotTimestamp
+            return transactionViewModel.getTimestamp() < 0
                     //you are valid if you are the genesis
                     && !Objects.equals(transactionViewModel.getHash(), Hash.NULL_HASH)
                     || transactionViewModel.getTimestamp() > (System.currentTimeMillis() / 1000) + MAX_TIMESTAMP_FUTURE;
         }
-        return transactionViewModel.getAttachmentTimestamp() < snapshotTimestampMs
+        return transactionViewModel.getAttachmentTimestamp() < 0
                 || transactionViewModel.getAttachmentTimestamp() > System.currentTimeMillis() + MAX_TIMESTAMP_FUTURE_MS;
     }
```
Comment (alongalky) [inline src/main/java/com/iota/iri/TransactionValidator.java:137]: Line 210: remove 'milestone' parameter in `checkSolidity`

File: src/main/java/com/iota/iri/TransactionValidator.java [hunk 7]
```diff
@@ -133,12 +126,12 @@ public int getMinWeightMagnitude() {
      */
     private boolean hasInvalidTimestamp(TransactionViewModel transactionViewModel) {
         if (transactionViewModel.getAttachmentTimestamp() == 0) {
-            return transactionViewModel.getTimestamp() < snapshotTimestamp
+            return transactionViewModel.getTimestamp() < 0
```
Comment (alongalky) [inline src/main/java/com/iota/iri/TransactionValidator.java:129]: Add comment explaining what this `< 0` means

File: src/main/java/com/iota/iri/TransactionValidator.java (hunk 6 above)
Comment (alongalky) [reply src/main/java/com/iota/iri/TransactionValidator.java:137]: Also in line 238

File: src/main/java/com/iota/iri/TransactionValidator.java (hunk 7 above)
Comment (alongalky) [reply src/main/java/com/iota/iri/TransactionValidator.java:129]: Maybe just update the comment above the function

File: src/main/java/com/iota/iri/TransactionValidator.java [hunk 8]
```diff
@@ -133,12 +126,12 @@ public int getMinWeightMagnitude() {
...
                     //you are valid if you are the genesis
                     && !Objects.equals(transactionViewModel.getHash(), Hash.NULL_HASH)
                     || transactionViewModel.getTimestamp() > (System.currentTimeMillis() / 1000) + MAX_TIMESTAMP_FUTURE;
         }
-        return transactionViewModel.getAttachmentTimestamp() < snapshotTimestampMs
+        return transactionViewModel.getAttachmentTimestamp() < 0
```
Comment (alongalky) [inline src/main/java/com/iota/iri/TransactionValidator.java:134]: same

**Comment by alon-e** [comment]
```
    if node_info['latestSolidSubtangleMilestone'] != all_nines:
KeyError: 'latestSolidSubtangleMilestone'
```
integration tests related to missing flags, from `iri-regression-tests`.
I'll first make the Travis changes and then rebase this.

**Comment by alon-e** [comment]
@alongalky - fixed.

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 9]
```diff
@@ -104,7 +95,7 @@ public JCommander parseConfigFromArgs(String[] args) throws ParameterException {
                 .acceptUnknownOptions(true)
                 .allowParameterOverwriting(true)
                 //This is the first line of JCommander Usage
-                .programName("java -jar iri-" + IRI.VERSION + ".jar")
+                .programName("java -jar iri-" +  IRI.VERSION + ".jar")
```
Comment (alongalky) [inline src/main/java/com/iota/iri/conf/BaseIotaConfig.java]: Stray space

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java [hunk 10]
```diff
@@ -451,114 +442,6 @@ public double getpPropagateRequest() {
...
-    public int getMilestoneStartIndex() {
-        return Defaults.MILESTONE_START_INDEX;
-    }
-
-    @Override
-    public int getNumberOfKeysInMilestone() {
-        return Defaults.NUM_KEYS_IN_MILESTONE;
-    }
-
     @Override
```
Comment (alongalky) [inline src/main/java/com/iota/iri/conf/BaseIotaConfig.java:434]: Lines 413-433 need to be removed as well, I believe

Comment (alongalky) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java:434]: Also 553:
```
    @Override
    public int getMilestoneStartIndex() {
        return Defaults.MILESTONE_START_INDEX;
    }
```

File: src/main/java/com/iota/iri/conf/TestnetConfig.java [hunk 11]
```diff
@@ -8,11 +8,6 @@
 
 public class TestnetConfig extends BaseIotaConfig {
 
-    protected String coordinator = Defaults.COORDINATOR_ADDRESS;
-    protected boolean dontValidateTestnetMilestoneSig = Defaults.DONT_VALIDATE_MILESTONE_SIG;
-    protected String snapshotFile = Defaults.SNAPSHOT_FILE;
-    protected String snapshotSignatureFile = Defaults.SNAPSHOT_SIG;
-    protected long snapshotTime = Defaults.SNAPSHOT_TIME;
     protected int mwm = Defaults.MWM;
     protected int milestoneStartIndex = Defaults.MILESTONE_START_INDEX;
```
Comment (alongalky) [inline src/main/java/com/iota/iri/conf/TestnetConfig.java]: remove

File: src/main/java/com/iota/iri/conf/TestnetConfig.java [hunk 12]
```diff
@@ -8,11 +8,6 @@
...
     protected int numberOfKeysInMilestone = Defaults.KEYS_IN_MILESTONE;
```
Comment (alongalky) [inline src/main/java/com/iota/iri/conf/TestnetConfig.java]: remove

File: src/main/java/com/iota/iri/conf/TestnetConfig.java [hunk 13]
```diff
@@ -160,13 +77,7 @@ public void setDbLogPath(String dbLogPath) {
...
-        String COORDINATOR_ADDRESS = "EQQFCZBIHRHWPXKMTOLMYUYPCN9XLMJPYZVFJSAY9FQHCCLWTOLLUGKKMXYFDBOOYFBLBI9WUEILGECYM";
-        boolean DONT_VALIDATE_MILESTONE_SIG = false;
-        String LOCAL_SNAPSHOTS_BASE_PATH = "testnet";
-        String SNAPSHOT_FILE = "/snapshotTestnet.txt";
         int REQUEST_HASH_SIZE = 49;
-        String SNAPSHOT_SIG = "/snapshotTestnet.sig";
-        int SNAPSHOT_TIME = 1522306500;
         int MWM = 9;
         int MILESTONE_START_INDEX = 434525;
         int KEYS_IN_MILESTONE = 22;
```
Comment (alongalky) [inline src/main/java/com/iota/iri/conf/TestnetConfig.java]: remove above 2 lines

File: src/main/java/com/iota/iri/network/Node.java [hunk 14]
```diff
@@ -270,7 +266,7 @@ private Runnable spawnNeighborDNSRefresherThread() {
      * a {@link TransactionViewModel} object from it and perform some basic validation
      * on the received transaction via  {@link TransactionValidator#runValidation}
      * 
-     * The packet is then added to  {@link receiveQueue} for further processing. 
+     * The packet is then added to  {@link receiveQueue} for further processing.
```
Comment (alongalky) [inline src/main/java/com/iota/iri/network/Node.java:269]: stray whitespace

File: src/main/java/com/iota/iri/network/Node.java [hunk 15]
```diff
@@ -627,7 +623,8 @@ private Runnable spawnTipRequesterThread() {
             while (!shuttingDown.get()) {
 
                 try {
-                    final TransactionViewModel transactionViewModel = TransactionViewModel.fromHash(tangle, milestoneTracker.latestMilestone);
+                    //TODO-CLIRI: send genesis request
```
Comment (alongalky) [inline src/main/java/com/iota/iri/network/Node.java]: Could you please elaborate on the comment + remove dash for consistency with the other TODOs?

File: src/main/java/com/iota/iri/network/Node.java [hunk 16]
```diff
@@ -540,7 +536,7 @@ public void replyToRequest(Hash requestedHash, Neighbor neighbor) {
     }
 
     private Hash getRandomTipPointer() throws Exception {
-        Hash tip = rnd.nextDouble() < configuration.getpSendMilestone() ? milestoneTracker.latestMilestone : tipsViewModel.getRandomSolidTipHash();
+        Hash tip = tipsViewModel.getRandomSolidTipHash();
         return tip == null ? Hash.NULL_HASH : tip;
     }
```
Comment (alongalky) [inline src/main/java/com/iota/iri/network/Node.java:542]: Line 570: `            Hash hash = transactionRequester.transactionToRequest(rnd.nextDouble() < configuration.getpSelectMilestoneChild());`

Probably need to kill it

File: src/main/java/com/iota/iri/network/Node.java (hunk 15 above)
Comment (alongalky) [reply src/main/java/com/iota/iri/network/Node.java]: Also fix comment in line 615

**Comment by alon-e** [comment]
tested (passed):
1. forming a local isolated tangle (1 node)
2. forming a local network and sending transactions around (2 nodes)

File: src/main/java/com/iota/iri/service/dto/GetNodeInfoResponse.java [hunk 17]
```diff
@@ -127,25 +93,18 @@
...
      * @return a {@link GetNodeInfoResponse} filled with all the provided parameters
      */
 	public static AbstractResponse create(String appName, String appVersion, int jreAvailableProcessors, long jreFreeMemory,
-	        String jreVersion, long maxMemory, long totalMemory, Hash latestMilestone, int latestMilestoneIndex,
-	        Hash latestSolidSubtangleMilestone, int latestSolidSubtangleMilestoneIndex, int milestoneStartIndex,
-	        int neighbors, int packetsQueueSize, long currentTimeMillis, int tips, 
-	        int numberOfTransactionsToRequest,  String[] features, String coordinatorAddress) {
+										  String jreVersion, long maxMemory, long totalMemory,
+										  int neighbors, int packetsQueueSize, long currentTimeMillis, int tips,
+										  int numberOfTransactionsToRequest, String[] features) {
```
Comment (alongalky) [inline src/main/java/com/iota/iri/service/dto/GetNodeInfoResponse.java]: weird indentation?

File: src/main/java/com/iota/iri/service/tipselection/impl/EntryPointSelectorGenesisImpl.java [hunk 18]
```diff
@@ -0,0 +1,21 @@
+package com.iota.iri.service.tipselection.impl;
+
+import com.iota.iri.model.Hash;
+import com.iota.iri.service.tipselection.EntryPointSelector;
+import com.iota.iri.storage.Tangle;
+
+/**
+ * Implementation of <tt>EntryPointSelector</tt> that given a depth N, returns a N-deep milestone.
```
Comment (alongalky) [inline src/main/java/com/iota/iri/service/tipselection/impl/EntryPointSelectorGenesisImpl.java]: Update comment

File: src/main/java/com/iota/iri/service/tipselection/impl/TipSelectorImpl.java [hunk 19]
```diff
@@ -71,37 +67,32 @@ public TipSelectorImpl(Tangle tangle,
...
-            if (!ledgerValidator.checkConsistency(tips)) {
-                throw new IllegalStateException(TIPS_NOT_CONSISTENT);
-            }
-
-            return tips;
-        } finally {
-            milestoneTracker.latestSnapshot.rwlock.readLock().unlock();
+
+        //preparation
+        Hash entryPoint = entryPointSelector.getEntryPoint(depth);
```
Comment (alongalky) [inline src/main/java/com/iota/iri/service/tipselection/impl/TipSelectorImpl.java:72]: remove `depth` argument? Could be added as a // TODO comment or issue

File: src/main/java/com/iota/iri/service/tipselection/impl/WalkValidatorImpl.java [hunk 20]
```diff
@@ -61,8 +58,9 @@ public boolean isValid(Hash transactionHash) throws Exception {
         } else if (!transactionViewModel.isSolid()) {
             log.debug("Validation failed: {} is not solid", transactionHash);
             return false;
+            //TODO-CLIRI remove belowMaxDepth, currently set to never trigger.
```
Comment (alongalky) [inline src/main/java/com/iota/iri/service/tipselection/impl/WalkValidatorImpl.java]: Remove // TODO comments in favor of github issues

Comment (alongalky) [reply src/main/java/com/iota/iri/service/tipselection/impl/WalkValidatorImpl.java]: Mention in issue also to remove the "max depth" config

File: src/main/java/com/iota/iri/TransactionValidator.java (hunk 6 above)
Comment (alon-e) [reply src/main/java/com/iota/iri/TransactionValidator.java:137]: good catch, I'll break that into another issue (and PR) #30

File: src/main/java/com/iota/iri/network/Node.java (hunk 16 above)
Comment (alon-e) [reply src/main/java/com/iota/iri/network/Node.java:542]: yup - part of #30

File: src/main/java/com/iota/iri/conf/BaseIotaConfig.java (hunk 10 above)
Comment (alon-e) [reply src/main/java/com/iota/iri/conf/BaseIotaConfig.java:434]: yup - will be part of #30

File: src/main/java/com/iota/iri/service/tipselection/impl/TipSelectorImpl.java (hunk 19 above)
Comment (alon-e) [reply src/main/java/com/iota/iri/service/tipselection/impl/TipSelectorImpl.java:72]: this is part of the `entryPoint` interface, I'm not sure it should be done yet - part of #31.

File: src/main/java/com/iota/iri/TransactionValidator.java (hunk 7 above)
Comment (alon-e) [reply src/main/java/com/iota/iri/TransactionValidator.java:129]: updated javadoc

File: src/main/java/com/iota/iri/network/Node.java (hunk 14 above)
Comment (alon-e) [reply src/main/java/com/iota/iri/network/Node.java:269]: ugh - always fixed by IDE formatter - it removes the stray whitespace

File: src/main/java/com/iota/iri/network/Node.java (hunk 16 above)
Comment (alon-e) [reply src/main/java/com/iota/iri/network/Node.java:542]: yup - as part of #30

File: src/main/java/com/iota/iri/conf/ProtocolConfig.java [hunk 21]
```diff
@@ -43,7 +41,6 @@
         String REQUEST_HASH_SIZE = "The size of the requested hash in a packet. Its size is derived from the minimal MWM value the network accepts. The larger the MWM -> the more trailing zeroes we can ignore -> smaller hash size.";
         String P_DROP_TRANSACTION = DescriptionHelper.PROB_OF + "dropping a received transaction. This is used only for testing purposes.";
         String P_SELECT_MILESTONE = DescriptionHelper.PROB_OF + "requesting a milestone transaction from a neighbor. This should be a large since it is imperative that we find milestones to get transactions confirmed";
```
Comment (alongalky) [inline src/main/java/com/iota/iri/conf/ProtocolConfig.java:43]: Since you deleted the line below, makes sense to remove this as well

**Review by alongalky — APPROVED** [review]

File: src/main/java/com/iota/iri/conf/ProtocolConfig.java (hunk 21 above)
Comment (alon-e) [reply src/main/java/com/iota/iri/conf/ProtocolConfig.java:43]: part of #30

**Pull request merged**